### PR TITLE
Objects

### PR DESCRIPTION
**Pull request by bfwetmore**
Object Lesson 

-Add Object Rough Work.
-Add Challenge for Objects.

### CODE REVIEW

File: objects/object-challenge/index.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+</head>
+<body>
+<h1>Pet Directory</h1>
+<main>
+    <!--
+          <h2>Joey</h2>
+          <h3>Dog | Australian Shepherd</h3>
+          <p>Age: 8</p>
+          <img src="img/aussie.jpg" alt="Australian Shepherd">
+    -->
```
Comment (do-work) [inline objects/object-challenge/index.html]: Its a good practice to remove comments before doing a PR.

File: objects/object-challenge/js/pets.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+const pets = [
+    {
+        name: 'Sam',
+        type: 'Dog',
+        breed: 'Australian Shepard',
+        age: 6,
+        photo: 'img/aussie.jpg'
```
Comment (do-work) [inline objects/object-challenge/js/pets.js]: What happens if the place you store images changes?
I like to see something like 'filename' == 'aussie.jpg' and then a const above this object where you define the path. 
Then in the spot you use it you can combine the two.

File: objects/object-challenge/js/pets.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        name: 'Tabby',
+        type: 'Cat',
+        breed: 'Mixed',
+        age: 2,
+        photo: 'img/tabby.jpg'
+    }
+];
+let html = '';
+for (let i = 0; i < pets.length; i++) {
```
Comment (do-work) [inline objects/object-challenge/js/pets.js]: Feel free to add some vertical spacing (aka an empty line between the object and the let - so line 45 can be empty.
Totally optional and its very much by feel.
Per `Clean Code` chapter5 on formatting one take is:
> openness separates concepts

Of course that is subjective too so YMMV.

File: objects/rough-work/index.html [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+<head>
+    <meta charset="utf-8">
+    <title>JavaScript Objects</title>
+    <link href="css/style.css" rel="stylesheet">
+</head>
+<body>
+<main></main>
+<script src="js/array-of-objects.js"></script>
+</body>
+</html>
```
Comment (do-work) [inline objects/rough-work/index.html]: Missing a blank line here.

File: objects/rough-work/js/array-of-objects.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
+// 1. Create an array to hold quiz questions and answers
```
Comment (do-work) [inline objects/rough-work/js/array-of-objects.js:1]: I prefer to remove comments like this.
Per Clean Code:
> Comments are not maintainable and prone to lying.
Stop and try to express via code.

File: objects/rough-work/js/array-of-objects.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    let question = questions[i].question;
+    let answer = questions[i].answer;
+    let response = prompt(question);
+
+    if (response === answer) {
+        correctAnswers++;
+        correct.push(question);
+    } else {
+        incorrect.push(question);
+    }
```
Comment (do-work) [inline objects/rough-work/js/array-of-objects.js:43]: Have you tried this one yet?
```
if (response === answer) {
    correctAnswers++;
    correct.push(question);
    continue;
}
incorrect.push(question);
```

**Review by do-work — CHANGES_REQUESTED** [review]

File: objects/rough-work/js/array-of-objects.js (hunk 6 above)
Comment (bfwetmore) [reply objects/rough-work/js/array-of-objects.js:43]: Yes I have tried the continue, it works great! Just following the instructor in my rough work.

File: objects/object-challenge/js/pets.js (hunk 3 above)
Comment (bfwetmore) [reply objects/object-challenge/js/pets.js]: I agree with this, I will do that from now on.

File: objects/rough-work/index.html (hunk 4 above)
Comment (bfwetmore) [reply objects/rough-work/index.html]: Fixed

File: objects/rough-work/js/array-of-objects.js (hunk 5 above)
Comment (bfwetmore) [reply objects/rough-work/js/array-of-objects.js:1]: Done

File: objects/object-challenge/index.html (hunk 1 above)
Comment (bfwetmore) [reply objects/object-challenge/index.html]: Removed

File: objects/object-challenge/js/pets.js (hunk 2 above)
Comment (bfwetmore) [reply objects/object-challenge/js/pets.js]: done

**Review by do-work — APPROVED** [review]

**Pull request merged**